game-core/src/main/java/games/strategy/engine/xml/TestAttachment.java

```diff
@@ -1,13 +1,22 @@
 package games.strategy.engine.xml;
 
+import java.util.Map;
+import java.util.function.Function;
+
+import com.google.common.collect.ImmutableMap;
+
 import games.strategy.engine.data.Attachable;
+import games.strategy.engine.data.AttachmentProperty;
 import games.strategy.engine.data.DefaultAttachment;
 import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.IAttachment;
 import games.strategy.engine.data.annotations.GameProperty;
 import games.strategy.engine.data.annotations.InternalDoNotExport;
 
 public class TestAttachment extends DefaultAttachment {
   private static final long serialVersionUID = 4886924951201479496L;
+  private static final Map<String, Function<IAttachment, AttachmentProperty<?>>> propertyMap = createPropertyMap();
```

Comment:
I presume we are storing the `propertyMap` to not recreate it? I would point out this is actually a memory leak, the map will never be GC'd, the reference to `TestAttachment` class handler will never be GC'd either. This is easy to fix by simply inlining. There is then less indirection. I'll note that that we do not have minimization of usage and declaration, going from line 18 we hop down to line 62. I was typically doing it from `getPropertyMap` on 69 up to 18, then back down to 62... We see a code cohesion code smell as 3 related entities are not all close together, one is split by 40 lines from the others

This is all solved very happily with a simple inlining : )
```
+  @Override
+  public Map<String, Function<IAttachment, AttachmentProperty<?>>> getPropertyMap() {
+    return ImmutableMap.<String, Function<IAttachment, AttachmentProperty<?>>>builder()
+        .put("value", ofCast(a -> AttachmentProperty.of(a::setValue, a::getValue, a::resetValue)))
+        .build();
+  }
+
+  private static Function<IAttachment, AttachmentProperty<?>> ofCast(
+      final Function<TestAttachment, AttachmentProperty<?>> function) {
+    return function.compose(TestAttachment.class::cast);
+  }
```

Finally at this point the `ofCast` would be nice to deal with. It's very slick and compact, but I'm afraid it is a bit too clever. I'm thinking we would just as well inline it, in-context it may make more sense even. Perhaps a better name or putting more into the helper function would help. I've generally found that the more method references you can convert lambda's to, the better you can clean up complex lambda chains, so that is the strategy I would then recommend if you look at that problem